MacOS compile error with -std=gnu89

I'm trying to setup the latest version of Zephyr on Mac OS 10.12.6 but I get the following error:

```
Scanning dependencies of target qconf_autogen
[  5%] Automatic MOC for target qconf
AutoMoc warning:
  "/Users/scott/Desktop/zephyr/scripts/kconfig/qconf.cc"
The file includes the moc file "qconf.moc", but does not contain a Q_OBJECT, Q_GADGET or Q_NAMESPACE macro.

[  5%] Built target qconf_autogen
Scanning dependencies of target qconf
[ 11%] Building CXX object kconfig/CMakeFiles/qconf.dir/qconf.cc.o
error: invalid argument '-std=gnu89' not allowed with 'C++/ObjC++'
make[2]: *** [kconfig/CMakeFiles/qconf.dir/qconf.cc.o] Error 1
make[1]: *** [kconfig/CMakeFiles/qconf.dir/all] Error 2
make: *** [all] Error 2
```

Removing `-std=gnu89` from `scripts/kconfig/CMakeLists.txt` did the trick.

--- a/scripts/kconfig/CMakeLists.txt
+++ b/scripts/kconfig/CMakeLists.txt
@@ -4,7 +4,6 @@ add_compile_options(
   -Wstrict-prototypes
   -O2
   -fomit-frame-pointer
-  -std=gnu89
   -Wno-format-security
   -DKBUILD_NO_NLS
 )

